### ISSUE TITLE
BigQuery date values not correct

### ISSUE THREAD

**Issue by VikramTiwari** [metabase/metabase#7435]
### Bugs

BigQuery based queries give an incorrect date when using `yesterday` filter on the date columns after day has passed in GMT.

- Your browser and the version: Chrome Latest
- Your operating system: OS X latest
- Your databases: BigQuery
- Metabase version: 0.28.6
- Metabase internal database: MySQL

- *Repeatable steps to reproduce the issue*
Create a query with yesterday filter. It works for whole day except in evenings until next day.

**Comment by VikramTiwari** [comment]
Related? https://github.com/metabase/metabase/issues/6636

**Comment by senior** [comment]
I've spent time trying to reproduce this issue and I'm not able to. As far as I know, BigQuery stores its dates/timestamps in GMT. From looking at the cod and running some tests, it looks to me like `yesterday` is getting properly converted to GMT, so if it is the next day in GMT, the query would show that as well.

Can you give the query that you're trying to run, the JVM timezone, report timezone (if you have one) and the expected vs. the actual results?

**Comment by VikramTiwari** [comment]
No changes to JVM timezone, server timezone is UTC:

```
$ date
Tue May 22 01:13:58 UTC 2018
```

Report timezone is set to US/Pacific but that shouldn't affect anything since BigQuery ignores it.

On making queries in UI using GUI query builder, here are the resultant SQL queries for last 30 days, without today and with today.

Without today:
```
SELECT sum([dataset_id.table_id.clicks]) AS [sum], CAST(date([dataset_id.table_id.date]) AS timestamp) AS [dataset_id.table_id.date]
FROM [dataset_id.table_id]
WHERE CAST(date([dataset_id.table_id.date]) AS timestamp) BETWEEN CAST(date(timestamp('2018-04-22T01:11:43.410Z')) AS timestamp)
   AND CAST(date(timestamp('2018-05-21T01:11:43.410Z')) AS timestamp)
GROUP BY [dataset_id.table_id.date]
ORDER BY [dataset_id.table_id.date] ASC
```

With today:
```
SELECT sum([dataset_id.table_id.clicks]) AS [sum], CAST(date([dataset_id.table_id.date]) AS timestamp) AS [dataset_id.table_id.date]
FROM [dataset_id.table_id]
WHERE CAST(date([dataset_id.table_id.date]) AS timestamp) BETWEEN CAST(date(timestamp('2018-04-22T01:11:12.246Z')) AS timestamp)
   AND CAST(date(current_timestamp()) AS timestamp)
GROUP BY [dataset_id.table_id.date]
ORDER BY [dataset_id.table_id.date] ASC
```

When I choose today as well, I should see the data for today as well but it's not there. This happens only between day change on GMT and day change on PDT (my location).

Let me know if I can help with more logs etc.

**Comment by senior** [comment]
This should be fixed by https://github.com/metabase/metabase/pull/7520. While working on a MySQL timezone bug, I found that on BigQuery we were incorrectly applying the report timezone, when we shouldn't. The fix has been merged and will be released with 0.30.0, would love to get confirmation that it fixes your issue.

**Comment by salsakran** [comment]
@camsaul can we confirm this is fixed, and if so closed this issue?

**Comment by camsaul** [comment]
Fix verified by test in #11060
